modules/lang-painless/src/main/java/org/elasticsearch/painless/ScriptRoot.java

```diff
@@ -72,4 +73,6 @@ public FunctionTable getFunctionTable() {
     public String getNextSyntheticName(String prefix) {
         return prefix + "$synthetic$" + syntheticCounter++;
     }
+
+    public void setDeterministic(boolean deterministic) { this.deterministic &= deterministic; }
```

Comment:
This method name is deceptive since it doesn't really set the deterministic flag. I wonder if this should be something like `markNonDeterministic()`, and does the set to false internally when that is called.

Reply:
Very nice. Done.